### PR TITLE
vSphere Cloud Provider: Fstype in storage class

### PR DESCRIPTION
**Pull request by abrarshivani**
This PR does following,

1. Adds fstype support in storage class for vSphere Cloud Provider.
2. Modify examples to include fstype in storage class.
3. Adds fstype support in storage class for Photon Controller Cloud Provider (@luomiao)

Internally reviewed [here](https://github.com/vmware/kubernetes/pull/88).

cc @pdhamdhere @tusharnt @kerneltime @BaluDontu @divyenpatel @luomiao 

```release-note
vSphere cloud provider: Allow specifying fstype in storage class.
```

### CODE REVIEW

**Comment by BaluDontu** [comment]
@k8s-bot bazel test this
@k8s-bot non-cri e2e test this
@k8s-bot verify test this
@k8s-bot kops aws e2e test this

**Comment by abrarshivani** [comment]
@k8s-bot bazel test this
@k8s-bot verify test this

**Comment by kerneltime** [comment]
@k8s-bot bazel test this
@k8s-bot cvm gce e2e test this
@k8s-bot verify test this

**Comment by kerneltime** [comment]
@abrarshivani please squash the commits.

**Comment by abrarshivani** [comment]
@jsafrane, Can you please review this PR?
//cc @kubernetes/sig-storage-pr-reviews

**Comment by abrarshivani** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot bazel test this
@k8s-bot verify test this

**Comment by jsafrane** [comment]
The code looks solid, please fix verification errors. `hack/verify-bazel.sh` is very unhappy about this PR.

**Comment by jsafrane** [comment]
Also, see #37801 for discussion about fsType in StorageClass.

File: test/e2e/vsphere_volume_fstype.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	8. Delete pod and Wait for Volume Disk to be detached from the Node.
+	9. Delete PVC, PV and Storage Class.
+*/
+
+var _ = framework.KubeDescribe("Volume fstype [Volume]", func() {
+	f := framework.NewDefaultFramework("volume-fstype")
+	var (
+		client    clientset.Interface
+		namespace string
+		nodeName  string
```
Comment (divyenpatel) [inline test/e2e/vsphere_volume_fstype.go]: remove nodeName if not used

File: test/e2e/vsphere_volume_fstype.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		if len(nodeList.Items) != 0 {
```
Comment (divyenpatel) [inline test/e2e/vsphere_volume_fstype.go]: Can we replace this with
```
	if (len(nodeList.Items) == 0) {
			framework.Failf("Unable to find ready and schedulable Node")
	}
```

**Comment by abrarshivani** [comment]
@k8s-bot kops aws e2e test this

File: test/e2e/vsphere_volume_fstype.go (hunk 2 above)
Comment (abrarshivani) [inline test/e2e/vsphere_volume_fstype.go]: Good Catch! Done.

**Comment by abrarshivani** [comment]
@jsafrane Verification errors are fixed now. Thanks, for pointing me to #37801 will keep an eye on it.

**Comment by abrarshivani** [comment]
@k8s-bot gci gce e2e test this

**Comment by abrarshivani** [comment]
@k8s-bot gci gce e2e test this
@k8s-bot non-cri e2e test this

**Pull request merged**